src/verification/others.rs

```diff
@@ -68,35 +89,40 @@ impl Verifier {
                     log::warn!("Some CA root certificates were ignored due to errors");
                 }
 
-                // While we load webpki-roots anyway, this can be helpful to know for troubleshooting.
                 if root_store.is_empty() {
                     log::error!("No CA certificates were loaded from the system");
+                } else {
+                    log::debug!("Loaded {added} CA certificates from the system");
                 }
 
-                // Finding TLS roots on Linux is not reliable enough to always depend on it
-                // across various distributions. Instead, we always load the WebPKI roots in
-                // addition so that a valid trust anchor is more likely to be available.
-                load_webpki_roots(&mut root_store);
-
-                log::debug!(
-                    "Loaded WebPKI roots in addition to {} roots from the system",
-                    added
-                );
+                #[cfg(feature = "extra-roots")]
+                {
+                    root_store.add_server_trust_anchors(&mut self.extra_roots.clone().into_iter());
```

Comment:
Nit: It would be nice to not need to duplicate these roots twice in memory for the lifetime of the verifier (three times if they're also hardcoded in the binary like `webpki-roots`. I'm not immediately sure to do this though since at this time we only have immutable access.

Reply:
I'll see if I can figure something out.

Reply:
I'm not sure I see a path forward here :-/ Perhaps with some adjustments on the rustls side to not require owned representations of trust anchors?

@complexspaces Do you think this is blocking for the rest of the work?

Reply:
I was thinking about this a little more, but couldn't we just wrap `extra_roots` in a `Mutex` and then use `Vec.drain` here? This shouldn't be in any kind of hot path since `init_verifier` is only called once per `Verifier` construction.

Reply:
> couldn't we just wrap extra_roots in a Mutex and then use Vec.drain here?

That worked great, thank you. I had tried using a `Mutex` previously but I think I confused myself trying to use `get_mut` instead of `try_lock` and bumping into the same requirement of `mut self` (silly in retrospect).

I've used an `if let Ok(mut extra_roots) = self.extra_roots.try_lock() { ... }` construction in my last push. Do you think that's reasonable or should we handle the `Err` case differently? Maybe failing initialization?

Reply:
Personally I would have just used `.unwrap()` because there's no way for the Mutex to get locked multiple times, so it could never be poisoned (which is the only route to a panic) but I'm fine with `if let Ok()` if others prefer that as long as we have a comment saying it can never fail.

Reply:
> Personally I would have just used .unwrap() because there's no way for the Mutex to get locked multiple times, so it could never be poisoned (which is the only route to a panic)

That makes sense to me. I'll switch to `unwrap` and add a comment.